### PR TITLE
Add a field `IsSkipped` to indicate if a script is skipped in PR checker. 

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

We already have the script to validate and trace the test scripts in PR checker. In this PR, we enhance this script, add a field to indicate if this script is recorded in our skip list. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
We already have the script to validate and trace the test scripts in PR checker. In this PR, we enhance this script, add a field to indicate if this script is recorded in our skip list. 

#### How did you do it?
Add a filed to indicate if the script is recorded in our skip list.

### CODE REVIEW

**Review by wangxin — APPROVED** [review]

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py [hunk 1]
```diff
@@ -165,9 +173,12 @@ def check_PRChecker_coverd(test_scripts, topology_type_pr_test_scripts):
             expanded_test_scripts.append(test_script)
 
     # Check if a script is included in the PR checker for the corresponding topology type
+    # And if this script is skipped in PR checker
     for test_script in expanded_test_scripts:
         topology_type = topo_name_to_type(test_script["topology"])
 
+        test_script["skipped"] = test_script["testscript"] in topology_type_pr_test_skip_scripts.get(topology_type, "")
```
Comment (opcoder0) [inline .azure-pipelines/testscripts_analyse/analyse_testscripts.py:180]: The comment at the top of the file shows the sample return value of this script can include now `skipped` as well.

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py [hunk 2]
```diff
@@ -145,10 +153,10 @@ def get_PRChecker_scripts():
             else:
                 topology_type_pr_test_scripts[topology_type] = set(value)
 
-    return topology_type_pr_test_scripts
+    return topology_type_pr_test_scripts, topology_type_pr_test_skip_scripts
 
 
-def check_PRChecker_coverd(test_scripts, topology_type_pr_test_scripts):
+def check_PRChecker_coverd(test_scripts, topology_type_pr_test_scripts, topology_type_pr_test_skip_scripts):
```
Comment (opcoder0) [inline .azure-pipelines/testscripts_analyse/analyse_testscripts.py:159]: Nit: I guess we could change the name to `expand_test_scripts` as it takes the current list and expands `any` type to all of the types in `PR_TOPOLOGY_TYPE`. And the function then returns `expanded_test_scripts` which is easier to understand. I feel `check_` functions usually represent returning status / boolean values.

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py [hunk 3]
```diff
@@ -130,8 +130,16 @@ def get_PRChecker_scripts():
         logging.error('Failed to load file {}, error {}'.format(f, e))
 
     topology_type_pr_test_scripts = {}
+    topology_type_pr_test_skip_scripts = {}
```
Comment (opcoder0) [inline .azure-pipelines/testscripts_analyse/analyse_testscripts.py:133]: Nit: Dictionaries can have a naming convention like `topology_type_pr_test_scripts` -> `test_scripts_per_topology_type` and `topology_type_pr_test_skip_scripts` -> `skipped_scripts_per_topology_type`

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py [hunk 4]
```diff
@@ -183,8 +194,9 @@ def upload_results(test_scripts):
 
 def main():
     test_scripts = collect_all_scripts()
-    topology_type_pr_test_scripts = get_PRChecker_scripts()
-    expanded_test_scripts = check_PRChecker_coverd(test_scripts, topology_type_pr_test_scripts)
+    topology_type_pr_test_scripts, topology_type_pr_test_skip_scripts = get_PRChecker_scripts()
```
Comment (opcoder0) [inline .azure-pipelines/testscripts_analyse/analyse_testscripts.py:197]: Nit: `get_PRChecker_scripts` can be renamed to `get_included_skipped_ts_by_topology` that way it also confirms to PEP 8.

**Review by opcoder0 — COMMENTED** [review]
Looks good. Only a bunch of minor suggestions.

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py (hunk 1 above)
Comment (yutongzhang-microsoft) [reply .azure-pipelines/testscripts_analyse/analyse_testscripts.py:180]: Good catch! Let me add it.

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py (hunk 4 above)
Comment (yutongzhang-microsoft) [reply .azure-pipelines/testscripts_analyse/analyse_testscripts.py:197]: Do you think `get_included_skipped_ts_by_topology_type` is much better?

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py (hunk 3 above)
Comment (yutongzhang-microsoft) [reply .azure-pipelines/testscripts_analyse/analyse_testscripts.py:133]: Thank you for suggestion, changed.

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py (hunk 2 above)
Comment (yutongzhang-microsoft) [reply .azure-pipelines/testscripts_analyse/analyse_testscripts.py:159]: Changed

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py (hunk 4 above)
Comment (opcoder0) [reply .azure-pipelines/testscripts_analyse/analyse_testscripts.py]: I think that is indeed very long. I am happy if you can make it shorter or choose another name altogether. I was only looking at removing the acronym caps from the name. `get_PRChecker_scripts` -> `get_pr_checker_scripts` is good too.

**Review by opcoder0 — APPROVED** [review]
LGTM!

File: .azure-pipelines/testscripts_analyse/analyse_testscripts.py (hunk 4 above)
Comment (yutongzhang-microsoft) [reply .azure-pipelines/testscripts_analyse/analyse_testscripts.py]: Let's use `get_pr_checker_scripts`

**Review by opcoder0 — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**